master/buildbot/test/unit/test_www_rest.py

```diff
@@ -80,6 +80,23 @@ def test_invalid_http_method(self):
         yield self.render_resource(self.rsrc, '/', method='PATCH')
         self.assertSimpleError('invalid HTTP method', 400)
 
+    def test_default_origin(self):
+        self.master.config.buildbotURL = 'http://server/path/'
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['http\\:\\/\\/server\\Z(?ms)'])
+
+        self.master.config.buildbotURL = 'http://server/'
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['http\\:\\/\\/server\\Z(?ms)'])
+
+        self.master.config.buildbotURL = 'http://server:8080/'
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['http\\:\\/\\/server\\:8080\\Z(?ms)'])
+
+        self.master.config.buildbotURL = 'https://server:8080/'
+        self.rsrc.reconfigResource(self.master.config)
+        self.assertEqual([r.pattern for r in self.rsrc.origins], ['https\\:\\/\\/server\\:8080\\Z(?ms)'])
```

Comment:
I usually just copy paste what assertEqual complains the value should be (after visually checking them)